src/config_builder.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Config builder for Loki Charmed Operator."""
+
+import os
+
+
+class ConfigBuilder:
+    """Loki configuration builder class.
+
+    Some minimal configuration is required for Loki to start, including: storage paths, schema,
+    ring.
+
+    Reference: https://grafana.com/docs/loki/latest/configuration/
+    """
+
+    _target: str = "all"
+    _auth_enabled: bool = False
+
+    def __init__(self, charm):
+        """Init method."""
+        self._charm = charm
```

Comment:
Should we replace it with c'tor args?
I'm inclined to having a dep-free config builder.

Reply:
c'tor args??

I also inclined to dep-free config builder.The problem here is that we need a lot of information from the charm to build the config, that why I'm injecting it.